shell/plugins/dashboard-store/__tests__/actions.spec.js

```diff
@@ -0,0 +1,125 @@
+import _actions from '@shell/plugins/dashboard-store/actions';
+
+const { findAll } = _actions;
+
+describe('dashboard-store: actions', () => {
+  let dispatchCalls = {};
+  let commitCalls = {};
+  // we're dumping our commits into a map so we can reference them
+  // actions and mutations are stored against their name in an array so we can capture if they're called multiple times
+  // arguments past the name of the action or mutation are stored in an array in case there are multiple
+  const commit = (mutation, ...args) => {
```

Comment:
All these properties must be defined in a `beforeEach`, letting them be instantiated each time and avoiding internal changes.

PS: If you have seen me doing it somewhere, it was my mistake as well 😅

Reply:
this would create it's own problems, partially because jest's mocks are globals and the calls to these methods are made asynchronously but it would also cause potential performance issues as this function is going to require quite a few tests and several assertions should be made for each combination of parameters provided to it. Jest's documentation prefers these mocks to be defined reset outside of a beforeEach.

https://github.com/jest-community/eslint-plugin-jest/blob/v24.4.0/docs/rules/no-hooks.md